reposilite-frontend/src/store/console/connection.js

```diff
@@ -67,37 +73,44 @@ export default function useConsole() {
     command.value = history.value[historyIdx.value]
   }
 
+  // this is needed to stop an error from appearing in console when
+  // switching/refreshing the page without closing the connection
+  // TODO: move somewhere else?
+  window.onbeforeunload = function () {
+    close();
+  };
+
   const onOpen = ref()
   const onMessage = ref()
   const onError = ref()
   const onClose = ref()
 
   const connect = (token) => {
     try {
-      connection.value = new WebSocket(consoleAddress)
+      connection.value = new Eventsource(consoleAddress, {
+        headers: {
+          Authorization: `xBasic ${btoa(`${token.name}:${token.secret}`)}`
+        },
+        // TODO: should we try to reconnect? how many attempts should be made?
+        disableRetry: true
```

Comment:
As far as I see, the SSE disconnects on errors. Retries help to recover from that, so I think it shouldn't be needed in our case then :thinking:

Reply:
I'm a bit confused by the wording of this. Do I leave it as is and just remove the TODO comment?

Reply:
I was just replying to the question from todo :sweat_smile: I think it's okay to disable retries and you can remove the comment :ok_hand: